proto/errors.go

```diff
@@ -23,6 +23,11 @@ import (
 	"github.com/cockroachdb/cockroach/util/retry"
 )
 
+// ErrorUnexpectedlySet is the string we panic on when we assert
+// `ResponseHeader.Error == nil` before calling
+// `ResponseHeader.SetGoError()`.
+const ErrorUnexpectedlySet = "error is unexpectedly set"
```

Comment:
It's weird to make a public constant for this. It's only used twice now, both in the storage package. We could make it a local constant in storage, or just use the same literal string in both places (it's not that important that it be exactly the same message)